rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -0,0 +1,727 @@
+            signer.init(true, privKey);
+            BigInteger[] components = signer.generateSignature(sighash.getBytes());
+            BtcECKey.ECDSASignature sig2 = new BtcECKey.ECDSASignature(components[0], components[1]).toCanonicalised();
+            bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeysToSignWith.get(0)), Lists.newArrayList(sig2.encodeToDER()), keccak256.getBytes());
+        }
+        if (privateKeysToSignWith.size() > 1) {
+            BtcECKey.ECDSASignature sig2 = privateKeysToSignWith.get(1).sign(sighash);
+            byte[] derEncodedSig2 = sig2.encodeToDER();
+            List derEncodedSigs2 = new ArrayList();
+            for (int i = 0; i < numberOfInputsToSign; i++) {
+                derEncodedSigs2.add(derEncodedSig2);
+            }
+            bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeysToSignWith.get(1)), derEncodedSigs2, keccak256.getBytes());
+        }
+        bridgeSupport.save();
+        track.commit();
+
+        provider = new BridgeStorageProvider(repository, PrecompiledContracts.BRIDGE_ADDR, bridgeConstantsRegtest, activationsBeforeForks);
+
+        if ("FullySigned".equals(expectedResult)) {
+            Assert.assertTrue(provider.getRskTxsWaitingForSignatures().isEmpty());
+            Assert.assertThat(logs, is(not(empty())));
+            Assert.assertThat(logs, hasSize(3));
+            LogInfo releaseTxEvent = logs.get(2);
+            Assert.assertThat(releaseTxEvent.getTopics(), hasSize(1));
+            Assert.assertThat(releaseTxEvent.getTopics(), hasItem(Bridge.RELEASE_BTC_TOPIC));
+            BtcTransaction releaseTx = new BtcTransaction(btcRegTestParams, ((RLPList) RLP.decode2(releaseTxEvent.getData()).get(0)).get(1).getRLPData());
+            Script retrievedScriptSig = releaseTx.getInput(0).getScriptSig();
+            Assert.assertEquals(4, retrievedScriptSig.getChunks().size());
+            assertTrue(retrievedScriptSig.getChunks().get(1).data.length > 0);
+            assertTrue(retrievedScriptSig.getChunks().get(2).data.length > 0);
+        } else {
+            Script retrievedScriptSig = provider.getRskTxsWaitingForSignatures().get(keccak256).getInput(0).getScriptSig();
+            Assert.assertEquals(4, retrievedScriptSig.getChunks().size());
+            boolean expectSignatureToBePersisted = "PartiallySigned".equals(expectedResult); // for "InvalidParameters"
+            Assert.assertEquals(expectSignatureToBePersisted, retrievedScriptSig.getChunks().get(1).data.length > 0);
+            assertFalse(retrievedScriptSig.getChunks().get(2).data.length > 0);
+        }
+    }
+
+    private BtcECKey findPublicKeySignedBy(List<BtcECKey> pubs, BtcECKey pk) {
+        for (BtcECKey pub : pubs) {
+            if (Arrays.equals(pk.getPubKey(), pub.getPubKey())) {
+                return pub;
+            }
+        }
+        return pk;
```

Comment:
This is a bit strange. If it can't find the public key that corresponds to the given private key, then it returns the private?

I think it would make more sense to return an Optional, what do you think? This way it may lead to confussion